src/node.rs

```diff
@@ -1 +1,148 @@
-pub struct Node;
+use conrod::{CustomWidget, CustomWidgetState};
+use conrod::{Color, Colorable};
+use conrod::Frameable;
+use conrod::{FontSize, Labelable};
+use conrod::{Mouse, MouseButtonState};
+use conrod::{Depth, Dimensions, Position, Positionable};
+use conrod::{UiId, Ui};
+use conrod::WidgetKind;
+
+#[derive(PartialEq, Copy, Clone, Debug)]
+pub enum NodeState {
+    Normal,
+    Dragging,
+}
+
+fn get_new_state(is_over: bool, prev: NodeState, mouse: Mouse) -> NodeState {
+    use self::NodeState::{Normal, Dragging};
+    match (is_over, prev, mouse.left) {
+        (true,  _,  MouseButtonState::Down) => Dragging,
+        _                                   => Normal,
+    }
+}
+
+impl NodeState {
+    /// Alter the widget color depending on the state.
+    fn color(&self, color: Color) -> Color {
+        match *self {
+            NodeState::Normal => color,
+            NodeState::Dragging => color.highlighted(),
+        }
+    }
+}
+
+impl CustomWidgetState for NodeState {
```

Comment:
You'll also now need to implement `init() -> Self` which returns the `NodeState`'s initial state.